src/scenes/level.js

```diff
@@ -506,4 +613,38 @@ export class LevelScene extends Phaser.Scene {
         this.autoClickers = [];
         this.clickObjects = [];
     }
+
+    updateSkillsText() {
+        let totalLevel = 0;
+
+        // Attack
+        let level = this.calcLevel(this.characterData.skills.attack);
+        this.attackText.text = level;
+        this.attackBottomText.text = level;
+        totalLevel += level;
+
+        // Ranged
+        level = this.calcLevel(this.characterData.skills.ranged);
+        this.rangedText.text = level;
+        this.rangedBottomText.text = level;
+        totalLevel += level;
+
+        // Magic
+        level = this.calcLevel(this.characterData.skills.magic);
+        this.magicText.text = level;
+        this.magicBottomText.text = level;
+        totalLevel += level;
+
+        this.totalLevelText.text = totalLevel;
+    }
+
+    calcLevel(xp, lv = 1) {
+        let currLvXp = 75*(Math.pow(1.104, lv));
+        if (xp > currLvXp) {
+            return (this.calcLevel(xp - currLvXp, lv+1));
```

Comment:
Should just inverse the equation and ceil(?) the result (watch out for negative levels in xps in lvl 1 range). Also I'd prefer to use the actual xp equation even if it is more complicated.

Reply:
Inverting it would just check if you have the xp difference at the current level, not if you have the total xp necessary. This could be circumvented by storing the level and xp separately and resetting xp to 0 once you level up, but I'm pretty sure runescape shows total xp on hover

Reply:
Use the rewritten summation without the truncation error: https://runescape.fandom.com/wiki/Experience#Equations

Reply:
Woops, will do when I'm home

Reply:
For future self:
```
function calcLevel(xp, lv = 1) {
    let curLvXp = Math.floor(.25*(lv + 300*Math.pow(2, lv/7)));
    if (xp > curLvXp) {
        return (this.calcLevel(xp - curLvXp, lv+1));
    }
    else {
        return lv;
    }
}
```

Reply:
Done

Reply:
If you use the L/8(L-1) + 75*(... ) + E(L) you should not need recursion.

Reply:
Oh you would actually need something more sophisticated. Alright I'm in favor of simple and stupid.